pkg/controllers/provisioning/suite_test.go

```diff
@@ -113,6 +114,44 @@ var _ = Describe("Provisioning", func() {
 		Expect(len(nodes.Items)).To(Equal(1))
 		ExpectScheduled(ctx, env.Client, pod)
 	})
+	It("should provision single pod if no other pod is received within the batch idle duration", func() {
+		ExpectApplied(ctx, env.Client, test.NodePool())
+		pod := test.UnschedulablePod()
+		ExpectApplied(ctx, env.Client, pod)
+		prov.Trigger(pod.UID)
+
+		wg := sync.WaitGroup{}
+		ExpectToWait(fakeClock, &wg)
+		result := ExpectSingletonReconciled(ctx, prov)
+		Expect(result.RequeueAfter).ToNot(BeNil())
+	})
+	It("should not extend the timeout if we receive the same pod past the batch idle duration", func() {
+		ctx = options.ToContext(ctx, test.Options(test.OptionsFields{
+			BatchMaxDuration:  lo.ToPtr(10 * time.Second),
+			BatchIdleDuration: lo.ToPtr(5 * time.Second),
+		}))
+		ExpectApplied(ctx, env.Client, test.NodePool())
+		pod := test.UnschedulablePod()
+		ExpectApplied(ctx, env.Client, pod)
+		prov.Trigger(pod.UID)
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		Expect(fakeClock.HasWaiters()).To(BeFalse())
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Eventually(func() bool { return fakeClock.HasWaiters() }).
+				// Caution: if another go routine takes longer than this timeout to
+				// wait on the clock, we will deadlock until the test suite timeout
+				WithTimeout(10 * time.Second).
+				WithPolling(10 * time.Millisecond).
+				Should(BeTrue())
+			fakeClock.Step(6 * time.Second)
```

Comment:
Do you want to just modify the ExpectToWait function to take in a time to step the clock? Or maybe just create an extension of it called `ExpectToWaitForDuration`